src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Runtime/TypeInfos/RuntimeArrayTypeInfo.cs

```diff
@@ -40,7 +40,11 @@ public sealed override int GetArrayRank()
 
         protected sealed override TypeAttributes GetAttributeFlagsImpl()
         {
-            return TypeAttributes.AutoLayout | TypeAttributes.AnsiClass | TypeAttributes.Class | TypeAttributes.Public | TypeAttributes.Sealed | TypeAttributes.Serializable;
+            TypeAttributes attrs = TypeAttributes.AutoLayout | TypeAttributes.AnsiClass | TypeAttributes.Class | TypeAttributes.Public | TypeAttributes.Sealed;
+#pragma warning disable SYSLIB0050 // Legacy serialization infrastructure is obsolete
+            attrs |= TypeAttributes.Serializable;
+#pragma warning restore SYSLIB0050
+            return attrs;
```

Comment:
I don't think the targeted `#pragma` is worth it here.

```suggestion
#pragma warning disable SYSLIB0050 // Legacy serialization infrastructure is obsolete
            return TypeAttributes.AutoLayout | TypeAttributes.AnsiClass | TypeAttributes.Class | TypeAttributes.Public | TypeAttributes.Sealed | TypeAttributes.Serializable;
#pragma warning restore SYSLIB0050
```